### ISSUE TITLE
File Object Specification

### ISSUE THREAD

**Issue by mpickering** [slackhq/slack-api-docs#5]
I'm a bit confused about which fields are always required for all the different objects. For example.. I received a file object with two fields not specified in the API documentation, `comments_count` and `ims`.

What does this snippet from the documentation mean?
> The name parameter may be null for unnamed files.

Does that mean it will be omitted for unnamed files or it will be the string `"null"`?

> The url property points to a publicly sharable URL directly to the file contents. Only editable-mode files currently have a url_download parameter. 

Does this mean that if `editable` is false then `url_download` will be omitted or empty (`""`)

Are any of the thumbnail fields guaranteed to exist?

Can you have a preview and a thumbnail?

In short, it would be greatly appreciated if someone could clarify which of the fields were optional and which were always present.

**Comment by mccreath** [comment]
Thanks for your patience. We're working on updating these docs, and we'll clarify all of these points in a future update.

To your specific questions:

* The `name` parameter would be included and the value would be the string `null`.
* If `editable` is set to `false`, the `usr_download` parameter is omitted from the object.
* Certain files might not have thumbnails (like snippets)
* The `preview` parameter is for text previews (the first few lines of a snippet in the channel, for example) and the `thumbnail` parameters are for images, so yes: both can exist on a file object.

If you have further questions, feel free to send them to feedback@slack.com. We're going to continue monitoring the issues here, but response time will generally be faster at that email address.
